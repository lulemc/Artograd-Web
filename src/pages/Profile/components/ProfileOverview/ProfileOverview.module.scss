.wrapper {
  background: #ffffff;
  padding: 24px 0;
}

.profileName {
  font-size: 24px;
  color: #0853c1;
  font-weight: 600;
  text-align: center;
  width: 100%;
}

.profileRole {
  font-size: 14px;
  width: 100%;
  text-align: center;
  margin: 0;
}

.avatar {
  cursor: pointer;
}

.divider {
  width: 100%;
  margin-top: 3px;
  margin-bottom: 3px;
  border-bottom: 1px solid var(--uui-divider);
}

.overviewSectionName {
  font-size: 14px;
  font-weight: 600;
  color: #111111;
}

.overviewSectionItem {
  font-size: 12px;
  font-weight: 600;
  text-align: left;
}
