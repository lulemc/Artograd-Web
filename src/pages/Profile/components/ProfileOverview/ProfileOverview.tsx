import styles from './ProfileOverview.module.scss';
import {
  Avatar,
  ErrorNotification,
  FlexCell,
  FlexRow,
  LinkButton,
  Panel,
  SuccessNotification,
  Text,
} from '@epam/uui';
import { useTranslation } from 'react-i18next';
import { useHistory } from 'react-router-dom';
import { RootState } from '../../../../store/store';
import { useSelector, useDispatch } from 'react-redux';
import { useUuiContext } from '@epam/uui-core';
import { ChangeProfileImageModal } from '../../modals/ChangeProfileImageModal/ChangeProfileImageModal';
import { createProfilePayload } from '../../../../services/helpers/profileHelper';
import { userApi } from '../../../../services/api/userAPI';
import {
  profileAvatarChanged,
  updateProfileInformation,
} from '../../../../store/slices/profileInformationSlice';
import { userAvatarChanged } from '../../../../store/identitySlice';
import { isPageLoading } from '../../../../store/helpersSlice';
import { svc } from '../../../../services';
import { useEffect } from 'react';
import {
  ACTIVE_TENDERS,
  OTHER_TENDERS,
  tenderApi,
} from '../../../../services/api/tenderApi';
import axios from 'axios';

export const ProfileOverview = () => {
  const { t } = useTranslation();
  const history = useHistory();

  useEffect(() => {
    axios
      .all([
        tenderApi.get(username, ACTIVE_TENDERS),
        tenderApi.get(username, OTHER_TENDERS),
      ])
      .then((response) => {
        const [activeTenders, others] = [response[0]?.data, response[1]?.data];

        dispatch(
          updateProfileInformation({
            activeTenders,
            others,
          }),
        );
      })
      .catch(() => null);
  }, []);

  const {
    picture,
    given_name,
    family_name,
    activeTenders,
    others,
    activeArtObjects,
    readyArtObjects,
  } = useSelector((state: RootState) => state.profileInformation);
  const organization = useSelector(
    (state: RootState) => state.profileInformation['custom:organization'],
  );
  const username = useSelector(
    (state: RootState) => state.identity.userData['cognito:username'],
  );
  const { uuiModals } = useUuiContext();
  const dispatch = useDispatch();

  const modalHandler = () => {
    uuiModals
      .show<string>((props) => (
        <ChangeProfileImageModal modalProps={props} picture={picture} />
      ))
      .then((picture: string) => {
        dispatch(isPageLoading(true));
        userApi
          .put(username, createProfilePayload({ picture: picture || '' }))
          .then(() => {
            dispatch(profileAvatarChanged({ picture: picture || '' }));
            dispatch(userAvatarChanged({ picture: picture || '' }));
            dispatch(isPageLoading(false));
            svc.uuiNotifications
              .show(
                (props) => (
                  <SuccessNotification {...props}>
                    <Text>{t('profilePage.Profile is updated')}</Text>
                  </SuccessNotification>
                ),
                { duration: 2 },
              )
              .catch(() => null);
          })
          .catch(() => {
            dispatch(isPageLoading(false));
            svc.uuiNotifications
              .show(
                (props) => (
                  <ErrorNotification {...props}>
                    <Text>{t('profilePage.Error in profile saving')}</Text>
                  </ErrorNotification>
                ),
                { duration: 2 },
              )
              .catch(() => null);
          });
      })
      .catch(() => null);
  };

  return (
    <Panel cx={styles.wrapper} shadow>
      <FlexRow justifyContent={'center'}>
        <Avatar
          img={picture}
          size={'90'}
          cx={styles.avatar}
          onClick={modalHandler}
        />
      </FlexRow>
      <FlexRow>
        <Text cx={styles.profileName}>
          {given_name} {family_name}
        </Text>
      </FlexRow>
      <FlexRow>
        <Text cx={styles.profileRole}>{organization}</Text>
      </FlexRow>
      <FlexRow>
        <div className={styles.divider} />
      </FlexRow>
      <FlexRow padding="24">
        <Text cx={styles.overviewSectionName}>
          {t('profilePage.My Tenders')}
        </Text>
      </FlexRow>
      <FlexRow padding="24">
        <FlexCell grow={3}>
          <LinkButton
            cx={styles.overviewSectionItem}
            rawProps={{ 'data-testid': `active-id` }}
            onClick={() =>
              history.push({ pathname: '/tenders', state: ACTIVE_TENDERS })
            }
            caption={t('profilePage.Active tenders')}
          />
        </FlexCell>
        <FlexCell grow={1} textAlign={'right'}>
          <Text cx={styles.overviewSectionItemValue}>{activeTenders}</Text>
        </FlexCell>
      </FlexRow>
      <FlexRow padding="24">
        <FlexCell grow={3}>
          <LinkButton
            cx={styles.overviewSectionItem}
            rawProps={{ 'data-testid': `others-id` }}
            onClick={() =>
              history.push({ pathname: '/tenders', state: OTHER_TENDERS })
            }
            caption={t('profilePage.Others')}
          />
        </FlexCell>
        <FlexCell grow={1} textAlign={'right'}>
          <Text cx={styles.overviewSectionItemValue}>{others}</Text>
        </FlexCell>
      </FlexRow>
      <FlexRow padding="24">
        <Text cx={styles.overviewSectionName}>
          {t('profilePage.My Art Objects')}
        </Text>
      </FlexRow>
      <FlexRow padding="24">
        <FlexCell grow={3}>
          <LinkButton
            cx={styles.overviewSectionItem}
            rawProps={{ 'data-testid': `active-art-id` }}
            onClick={() => history.push('/home')}
            caption={t('profilePage.Active art objects')}
          />
        </FlexCell>
        <FlexCell grow={1} textAlign={'right'}>
          <Text cx={styles.overviewSectionItemValue}>{activeArtObjects}</Text>
        </FlexCell>
      </FlexRow>
      <FlexRow padding="24">
        <FlexCell grow={3}>
          <LinkButton
            cx={styles.overviewSectionItem}
            rawProps={{ 'data-testid': `ready-id` }}
            onClick={() => history.push('/home')}
            caption={t('profilePage.Ready art objects')}
          />
        </FlexCell>
        <FlexCell grow={1} textAlign={'right'}>
          <Text cx={styles.overviewSectionItemValue}>{readyArtObjects}</Text>
        </FlexCell>
      </FlexRow>
    </Panel>
  );
};
