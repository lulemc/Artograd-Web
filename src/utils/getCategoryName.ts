import { CategoryItemType } from '../types';

export const categoryList: CategoryItemType[] = [
  { id: 0, name: 'tendersPages.newTender.categories.sculptures' },
  { id: 1, name: 'tendersPages.newTender.categories.mosaics' },
  { id: 2, name: 'tendersPages.newTender.categories.murals' },
  { id: 3, name: 'tendersPages.newTender.categories.graffiti' },
  { id: 4, name: 'tendersPages.newTender.categories.functionalArt' },
  {
    id: 5,
    name: 'tendersPages.newTender.categories.interactiveInstallations',
  },
  { id: 6, name: 'tendersPages.newTender.categories.botanicalArt' },
  { id: 7, name: 'tendersPages.newTender.categories.waterFeatures' },
  { id: 8, name: 'tendersPages.newTender.categories.themedGardens' },
  { id: 9, name: 'tendersPages.newTender.categories.recycled' },
];

export const getCategoryName = (id: string) => {
  return categoryList.find((category) => category.id === Number(id));
};
