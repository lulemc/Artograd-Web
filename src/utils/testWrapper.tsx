import { renderWithContextAsync } from '@epam/uui-test-utils';
import { MemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import { RootState, rootReducer, store } from '../store/store';
import { MemoryRouter } from 'react-router-dom';
import { createStore } from '@reduxjs/toolkit';
import { initialState as identityState } from '../store/identitySlice';
import { initialState as helpersState } from '../store/helpersSlice';
import { initialState as profileInformationReducer } from '../store/slices/profileInformationSlice';
import { initialState as profileFundrasingReducer } from '../store/slices/profileFundrasingSlice';
import { PersistPartial } from 'redux-persist/lib/persistReducer';

export const initialState: RootState & PersistPartial = {
  identity: identityState,
  helpers: helpersState,
  profileInformation: profileInformationReducer,
  profileFundrasing: profileFundrasingReducer,
  _persist: {
    version: 1,
    rehydrated: true,
  },
};

export const testWrapper = async ({
  component,
  history,
  path,
  state = initialState,
}: {
  component: React.JSX.Element;
  history: MemoryHistory<unknown>;
  path?: string;
  state?: RootState;
}) => {
  return path ? (
    <Provider store={store}>
      <MemoryRouter initialEntries={[path]}>{component}</MemoryRouter>
    </Provider>
  ) : (
    renderWithContextAsync(
      <Provider store={createStore(rootReducer, state)}>
        <Router history={history}>{component}</Router>
      </Provider>,
    )
  );
};
