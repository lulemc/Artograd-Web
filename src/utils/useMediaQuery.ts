import { useEffect, useState } from 'react';

export enum DeviceType {
  mobile = '(max-width: 767px)',
  tablet = '(max-width: 1023px)',
  laptop = '(min-width: 1024px)',
}

export const useMediaQuery = (query: DeviceType) => {
  const mediaMatch = window.matchMedia(query);
  const [matches, setMatches] = useState(mediaMatch.matches);
  useEffect(() => {
    const handler = (e: { matches: boolean }) => setMatches(e.matches);
    mediaMatch.addListener(handler);
    return () => mediaMatch.removeListener(handler);
  });
  return matches;
};
