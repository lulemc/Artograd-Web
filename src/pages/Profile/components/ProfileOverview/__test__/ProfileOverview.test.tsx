import { createMemoryHistory } from 'history';
import { ProfileOverview } from '../ProfileOverview';
import { testWrapper } from '../../../../../utils/testWrapper';
import { fireEvent, screen } from '@epam/uui-test-utils';

describe('Profile Overview', () => {
  const history = createMemoryHistory();

  test('Should create the profile overview component', async () => {
    const component = await testWrapper({
      component: <ProfileOverview />,
      history,
    });
    expect(component).toMatchSnapshot();
  });

  test('Should redirect by clicking Others', async () => {
    await testWrapper({ component: <ProfileOverview />, history });

    fireEvent.click(screen.getByTestId('others-id'));

    expect(history.location.pathname).toBe('/tenders');
  });

  test('Should redirect by clicking Active tenders', async () => {
    await testWrapper({ component: <ProfileOverview />, history });

    fireEvent.click(screen.getByTestId('active-id'));

    expect(history.location.pathname).toBe('/tenders');
  });

  test('Should redirect by clicking Active Art', async () => {
    await testWrapper({ component: <ProfileOverview />, history });

    fireEvent.click(screen.getByTestId('active-art-id'));

    expect(history.location.pathname).toBe('/home');
  });

  test('Should redirect by clicking Ready tenders', async () => {
    await testWrapper({ component: <ProfileOverview />, history });

    fireEvent.click(screen.getByTestId('ready-id'));

    expect(history.location.pathname).toBe('/home');
  });

  test('Should redirect by clicking Ready tenders', async () => {
    await testWrapper({ component: <ProfileOverview />, history });

    fireEvent.click(screen.getByTestId('ready-id'));

    expect(history.location.pathname).toBe('/home');
  });
});
