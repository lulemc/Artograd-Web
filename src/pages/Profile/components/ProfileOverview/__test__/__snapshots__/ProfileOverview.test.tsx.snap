// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Overview Should create the profile overview component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="uui-panel nNCe-- fI3ZAq wrapper uEZmHy"
      >
        <div
          class="uui-flex-row kfXZUl _7f-LqE _5Jq2IB RkwqFu IO7lfg"
        >
          <img
            class="hd78KB avatar"
            height="90"
            src="https://static.cdn.epam.com/uploads/690afa39a93c88c4dd13758fe1d869d5/EPM-UUI/Images/avatar_placeholder.jpg"
            width="90"
          />
        </div>
        <div
          class="uui-flex-row kfXZUl _7f-LqE _5Jq2IB RkwqFu"
        >
          <div
            class="Ka60RD uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography CMzhku xhqm1P jzu7jV profileName doyi1y"
          >
             
          </div>
        </div>
        <div
          class="uui-flex-row kfXZUl _7f-LqE _5Jq2IB RkwqFu"
        >
          <div
            class="Ka60RD uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography CMzhku xhqm1P jzu7jV profileRole doyi1y"
          />
        </div>
        <div
          class="uui-flex-row kfXZUl _7f-LqE _5Jq2IB RkwqFu"
        >
          <div
            class="divider"
          />
        </div>
        <div
          class="uui-flex-row kfXZUl _7f-LqE QjrJq4 _5Jq2IB RkwqFu"
        >
          <div
            class="Ka60RD uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography CMzhku xhqm1P jzu7jV overviewSectionName doyi1y"
          >
            My Tenders
          </div>
        </div>
        <div
          class="uui-flex-row kfXZUl _7f-LqE QjrJq4 _5Jq2IB RkwqFu"
        >
          <div
            class=""
            style="min-width: 0; flex-grow: 3; flex-basis: 0px;"
          >
            <button
              class="uui-button-box uui-enabled -clickable uui-link_button iTxfW2 uui-size-36 uui-no-left-icon uui-no-right-icon uui-color-primary"
              data-testid="active-id"
              tabindex="0"
              type="button"
            >
              <div
                class="uui-caption"
              >
                Active tenders
              </div>
            </button>
          </div>
          <div
            class=""
            style="min-width: 0; flex-grow: 1; flex-basis: 0px; text-align: right;"
          >
            <div
              class="Ka60RD uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography CMzhku xhqm1P jzu7jV overviewSectionItemValue doyi1y"
            >
              0
            </div>
          </div>
        </div>
        <div
          class="uui-flex-row kfXZUl _7f-LqE QjrJq4 _5Jq2IB RkwqFu"
        >
          <div
            class=""
            style="min-width: 0; flex-grow: 3; flex-basis: 0px;"
          >
            <button
              class="uui-button-box uui-enabled -clickable uui-link_button iTxfW2 uui-size-36 uui-no-left-icon uui-no-right-icon uui-color-primary"
              data-testid="others-id"
              tabindex="0"
              type="button"
            >
              <div
                class="uui-caption"
              >
                Others
              </div>
            </button>
          </div>
          <div
            class=""
            style="min-width: 0; flex-grow: 1; flex-basis: 0px; text-align: right;"
          >
            <div
              class="Ka60RD uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography CMzhku xhqm1P jzu7jV overviewSectionItemValue doyi1y"
            >
              0
            </div>
          </div>
        </div>
        <div
          class="uui-flex-row kfXZUl _7f-LqE QjrJq4 _5Jq2IB RkwqFu"
        >
          <div
            class="Ka60RD uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography CMzhku xhqm1P jzu7jV overviewSectionName doyi1y"
          >
            My Art Objects
          </div>
        </div>
        <div
          class="uui-flex-row kfXZUl _7f-LqE QjrJq4 _5Jq2IB RkwqFu"
        >
          <div
            class=""
            style="min-width: 0; flex-grow: 3; flex-basis: 0px;"
          >
            <button
              class="uui-button-box uui-enabled -clickable uui-link_button iTxfW2 uui-size-36 uui-no-left-icon uui-no-right-icon uui-color-primary"
              data-testid="active-art-id"
              tabindex="0"
              type="button"
            >
              <div
                class="uui-caption"
              >
                Active art objects
              </div>
            </button>
          </div>
          <div
            class=""
            style="min-width: 0; flex-grow: 1; flex-basis: 0px; text-align: right;"
          >
            <div
              class="Ka60RD uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography CMzhku xhqm1P jzu7jV overviewSectionItemValue doyi1y"
            >
              0
            </div>
          </div>
        </div>
        <div
          class="uui-flex-row kfXZUl _7f-LqE QjrJq4 _5Jq2IB RkwqFu"
        >
          <div
            class=""
            style="min-width: 0; flex-grow: 3; flex-basis: 0px;"
          >
            <button
              class="uui-button-box uui-enabled -clickable uui-link_button iTxfW2 uui-size-36 uui-no-left-icon uui-no-right-icon uui-color-primary overviewSectionItem"
              data-testid="ready-id"
              tabindex="0"
              type="button"
            >
              <div
                class="uui-caption"
              >
                Ready art objects
              </div>
            </button>
          </div>
          <div
            class=""
            style="min-width: 0; flex-grow: 1; flex-basis: 0px; text-align: right;"
          >
            <div
              class="Ka60RD uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography CMzhku xhqm1P jzu7jV overviewSectionItemValue doyi1y"
            >
              0
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="uui-panel nNCe-- fI3ZAq wrapper uEZmHy"
    >
      <div
        class="uui-flex-row kfXZUl _7f-LqE _5Jq2IB RkwqFu IO7lfg"
      >
        <img
          class="hd78KB avatar"
          height="90"
          src="https://static.cdn.epam.com/uploads/690afa39a93c88c4dd13758fe1d869d5/EPM-UUI/Images/avatar_placeholder.jpg"
          width="90"
        />
      </div>
      <div
        class="uui-flex-row kfXZUl _7f-LqE _5Jq2IB RkwqFu"
      >
        <div
          class="Ka60RD uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography CMzhku xhqm1P jzu7jV profileName doyi1y"
        >
           
        </div>
      </div>
      <div
        class="uui-flex-row kfXZUl _7f-LqE _5Jq2IB RkwqFu"
      >
        <div
          class="Ka60RD uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography CMzhku xhqm1P jzu7jV profileRole doyi1y"
        />
      </div>
      <div
        class="uui-flex-row kfXZUl _7f-LqE _5Jq2IB RkwqFu"
      >
        <div
          class="divider"
        />
      </div>
      <div
        class="uui-flex-row kfXZUl _7f-LqE QjrJq4 _5Jq2IB RkwqFu"
      >
        <div
          class="Ka60RD uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography CMzhku xhqm1P jzu7jV overviewSectionName doyi1y"
        >
          My Tenders
        </div>
      </div>
      <div
        class="uui-flex-row kfXZUl _7f-LqE QjrJq4 _5Jq2IB RkwqFu"
      >
        <div
          class=""
          style="min-width: 0; flex-grow: 3; flex-basis: 0px;"
        >
          <button
            class="uui-button-box uui-enabled -clickable uui-link_button iTxfW2 uui-size-36 uui-no-left-icon uui-no-right-icon uui-color-primary"
            data-testid="active-id"
            tabindex="0"
            type="button"
          >
            <div
              class="uui-caption"
            >
              Active tenders
            </div>
          </button>
        </div>
        <div
          class=""
          style="min-width: 0; flex-grow: 1; flex-basis: 0px; text-align: right;"
        >
          <div
            class="Ka60RD uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography CMzhku xhqm1P jzu7jV overviewSectionItemValue doyi1y"
          >
            0
          </div>
        </div>
      </div>
      <div
        class="uui-flex-row kfXZUl _7f-LqE QjrJq4 _5Jq2IB RkwqFu"
      >
        <div
          class=""
          style="min-width: 0; flex-grow: 3; flex-basis: 0px;"
        >
          <button
            class="uui-button-box uui-enabled -clickable uui-link_button iTxfW2 uui-size-36 uui-no-left-icon uui-no-right-icon uui-color-primary"
            data-testid="others-id"
            tabindex="0"
            type="button"
          >
            <div
              class="uui-caption"
            >
              Others
            </div>
          </button>
        </div>
        <div
          class=""
          style="min-width: 0; flex-grow: 1; flex-basis: 0px; text-align: right;"
        >
          <div
            class="Ka60RD uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography CMzhku xhqm1P jzu7jV overviewSectionItemValue doyi1y"
          >
            0
          </div>
        </div>
      </div>
      <div
        class="uui-flex-row kfXZUl _7f-LqE QjrJq4 _5Jq2IB RkwqFu"
      >
        <div
          class="Ka60RD uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography CMzhku xhqm1P jzu7jV overviewSectionName doyi1y"
        >
          My Art Objects
        </div>
      </div>
      <div
        class="uui-flex-row kfXZUl _7f-LqE QjrJq4 _5Jq2IB RkwqFu"
      >
        <div
          class=""
          style="min-width: 0; flex-grow: 3; flex-basis: 0px;"
        >
          <button
            class="uui-button-box uui-enabled -clickable uui-link_button iTxfW2 uui-size-36 uui-no-left-icon uui-no-right-icon uui-color-primary"
            data-testid="active-art-id"
            tabindex="0"
            type="button"
          >
            <div
              class="uui-caption"
            >
              Active art objects
            </div>
          </button>
        </div>
        <div
          class=""
          style="min-width: 0; flex-grow: 1; flex-basis: 0px; text-align: right;"
        >
          <div
            class="Ka60RD uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography CMzhku xhqm1P jzu7jV overviewSectionItemValue doyi1y"
          >
            0
          </div>
        </div>
      </div>
      <div
        class="uui-flex-row kfXZUl _7f-LqE QjrJq4 _5Jq2IB RkwqFu"
      >
        <div
          class=""
          style="min-width: 0; flex-grow: 3; flex-basis: 0px;"
        >
          <button
            class="uui-button-box uui-enabled -clickable uui-link_button iTxfW2 uui-size-36 uui-no-left-icon uui-no-right-icon uui-color-primary overviewSectionItem"
            data-testid="ready-id"
            tabindex="0"
            type="button"
          >
            <div
              class="uui-caption"
            >
              Ready art objects
            </div>
          </button>
        </div>
        <div
          class=""
          style="min-width: 0; flex-grow: 1; flex-basis: 0px; text-align: right;"
        >
          <div
            class="Ka60RD uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography CMzhku xhqm1P jzu7jV overviewSectionItemValue doyi1y"
          >
            0
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByAria": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByRoleAndText": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByAria": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByRoleAndText": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByAria": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByRoleAndText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByAria": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByRoleAndText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByAria": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByRoleAndText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByAria": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByRoleAndText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
